DigitalLearningSolutions.Web/Controllers/LoginController.cs

```diff
@@ -40,37 +41,84 @@ public IActionResult Index(LoginViewModel model)
                 return View("Index", model);
             }
 
-            var (unverifiedAdminUser, unverifiedDelegateUsers) = userService.GetUsersByUsername(model.Username);
+            var (adminUser, delegateUsers) = userService.GetUsersByUsername(model.Username);
 
-            if (unverifiedAdminUser == null && unverifiedDelegateUsers.Count == 0)
+            if (adminUser == null && delegateUsers.Count == 0)
             {
                 ModelState.AddModelError("Username", "No account with that email address or user ID could be found.");
                 return View("Index", model);
             }
 
             var (verifiedAdminUser, verifiedDelegateUsers) =
-                loginService.VerifyUsers(model.Password, unverifiedAdminUser, unverifiedDelegateUsers);
+                loginService.VerifyUsers(model.Password, adminUser, delegateUsers);
 
             if (verifiedAdminUser == null && verifiedDelegateUsers.Count == 0)
             {
                 ModelState.AddModelError("Password", "The password you have entered is incorrect.");
                 return View("Index", model);
             }
 
-            var approvedDelegateUser = verifiedDelegateUsers.FirstOrDefault(du => du.Approved);
+            var approvedDelegateUsers = verifiedDelegateUsers.Where(du => du.Approved).ToList();
 
-            if (verifiedAdminUser == null && approvedDelegateUser == null)
+            if (verifiedAdminUser == null && !approvedDelegateUsers.Any())
             {
                 return View("AccountNotApproved");
             }
 
-            verifiedAdminUser ??= loginService.GetVerifiedAdminUserAssociatedWithDelegateUser(verifiedDelegateUsers.First(), model.Password);
-            
-            LogIn(verifiedAdminUser, approvedDelegateUser, model.RememberMe);
-            return RedirectToAction("Index", "Home");
+            verifiedAdminUser ??=
+                loginService.GetVerifiedAdminUserAssociatedWithDelegateUser(verifiedDelegateUsers.First(), model.Password);
+
+            var availableCentres = GetAvailableCentres(verifiedAdminUser, approvedDelegateUsers);
+
+            if (availableCentres.Count == 1)
+            {
+                return LogIn(verifiedAdminUser, approvedDelegateUsers.FirstOrDefault(), model.RememberMe);
+            }
+
+            TempData.Clear();
+            TempData["RememberMe"] = model.RememberMe;
+            TempData.Set(adminUser);
+            TempData.Set(approvedDelegateUsers);
+
+            return View("ChooseACentre", availableCentres);

```

Comment:
I'd have a minor preference of having ChooseACentreViewModel with a field userDetailsInCentre[] or something similar. (so instead of passing the array to the view, we pass a viewmodel that has a field for this array. Then, if we ever need to add any other properties, it'll be easy to do :)